contracts/sol6/KyberNetwork.sol

```diff
@@ -939,18 +865,15 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
 
         reservesData.srcAmounts = new uint256[](numReserves);
         reservesData.rates = new uint256[](numReserves);
-        reservesData.addresses = new IKyberReserve[](numReserves);
         feesAccountedDestBps = new uint256[](numReserves);
+        reservesData.addresses = kyberStorage.getListReservesByIds(reservesData.ids);
```

Comment:
better do this in line 820
update that function name from getFeeAccountedAndEntitledRebateData
to 
getReserveData

And query all data in 1 call. save a contract call.

Reply:
Updated, change name to getReserveData